CONTRIBUTING.md

```diff
@@ -1,39 +1,123 @@
 
-### Code reviews
+### Pull Request Guidelines
 
-All submissions, including submissions by project members, require review. We
-use Github pull requests for this purpose.
+* Search [GitHub](https://github.com/firebase/firebase-js-sdk/pulls) for an open or closed Pull Request that relates to your submission. You don't want to duplicate effort.
+* Create an issue to discuss a change before submitting a PR. We'd hate to have to turn down your contributions because of something that could have been communicated early on.
+* Make your changes in a new git branch:
 
-### The small print
+  ```shell
+  git checkout -b my-fix-branch master
+  ```
 
-Contributions made by corporations are covered by a different agreement than the
-one above, the [Software Grant and Corporate Contributor License
-Agreement](https://cla.developers.google.com/about/google-corporate).
+* Create your patch, **including appropriate test cases**. Patches with tests are more likely to be merged.
+* Avoid checking in files that shouldn't be tracked (e.g `node_modules`, `gulp-cache`, `.tmp`, `.idea`). We recommend using a [global](#global-gitignore) gitignore for this.
+* Commit your changes using a commit message that follows our [commit message guidelines](#commit-message-guidelines).
 
-### Commit messages
+     ```shell
+     git commit -a
+     ```
+  _Note: the optional commit `-a` command line option will automatically "add" and "rm" edited files._
+
+* Test your changes locally to ensure everything is in good working order:
+
+    ```shell
+   npm test
+    ```
+
+* Push your branch to GitHub:
+
+    ```shell
+    git push origin my-fix-branch
+    ```
+
+* In GitHub, send a pull request to `firebase-js-sdk:master`.
+* All pull requests must be reviewed by a member of the Firebase JS SDK team who will merge it when/if they feel it is good to go.
+
+That's it! Thank you for your contribution!
+
+## <a name="commit-message-guidelines"></a> Commit Message Guidelines
 
 This repository follows the commit message format defined by the 
 [validate-commit-msg](https://npm.im/validate-commit-msg) package on NPM. This is
 to make the git history easy to follow, and make it easier to identify which
 commits are associated with features, bugfixes, etc.
 
 We are also Commitizen friendly! If you have the [Commitizen CLI](https://npm.im/commitizen) installed
```

Comment:
Can't see exactly what you're referring to, the phrase "CLI" only occurs once in the whole doc.